tests/e2e/cucumber/features/smoke/link.feature

```diff
@@ -63,3 +63,26 @@ Feature: link
       | resource  | type |
       | lorem.txt | file |
     And "Alice" logs out
+
+
+  Scenario: public link for folder (by authenticated user)
+    Given "Admin" creates following user using API
+      | id    |
+      | Brian |
+    And "Alice" logs in
+    And "Alice" creates the following resources
+      | resource     | type   |
+      | folderPublic | folder |
+    And "Alice" uploads the following resources
+      | resource  | to           |
+      | lorem.txt | folderPublic |
+    And "Alice" shares the following resources using the sidebar panel
+      | resource     | recipient | type | role     |
+      | folderPublic | Brian     | user | Can view |
+    And "Alice" creates a public link for the resource "folderPublic" using the sidebar panel
+    And "Alice" renames the most recently created public link of resource "folderPublic" to "folderLink"
+    And "Alice" logs out
+    And "Brian" logs in
+    When "Brian" opens the public link "folderLink"
+    Then "Brian" should see file "lorem.txt" but should not be able to edit
+    And "Brian" logs out
```

Comment:
what about:
- Alice shares with editor role to Brian
- Alice creates public link (viewer role)
- Brian opens the link
- Brain can edit the file (because he has editor role in the folder)